modules/n1ql/pages/n1ql-language-reference/insert.adoc

```diff
@@ -137,25 +137,81 @@ If you assign an alias to the keyspace reference, the `AS` keyword may be omitte
 ----
 
 image::n1ql-language-reference/insert-values.png["( '(' 'PRIMARY'? 'KEY' ',' 'VALUE' ')' )? values-clause"]
 
 Specifies one or more documents to be inserted using the VALUES clause.
 Each document requires a unique key and the values must be specified as a well-formed JSON document.
 
-NOTE: The keyword PRIMARY is optional.
-There is no syntactic difference between PRIMARY KEY and KEY.
+The bracketed KEY and VALUE keywords are purely a visual mnemonic to indicate that you are setting the key and value for the inserted document.
+There is no syntactic requirement to include these keywords when using the Insert Values syntax.
+Also note that there is no syntactic difference between PRIMARY KEY and KEY.
 
-The document key should be unique within the Couchbase bucket.
+****
+[.status]#Couchbase Server 6.5.1#
+
+In Couchbase Server 6.5.1, the Insert Values syntax supports an optional third keyword `OPTIONS`.
+
+[subs="normal"]
+----
+insert-values ::= [ "(" [ PRIMARY ] KEY "," VALUE [ "," OPTIONS ] ")" ] <<values-clause,values-clause>>
+----
+
+image::n1ql-language-reference/insert-values-651.png["( '(' 'PRIMARY'? 'KEY' ',' 'VALUE' ( ',' 'OPTIONS' )? ')' )? values-clause"]
+
+Again, the `OPTIONS` keyword is purely a visual mnemonic to indicate that you are setting metadata for the inserted document.
+There is no syntactic requirement to include the `OPTIONS` keyword when setting metadata for the inserted document.
+****
+
+[[values-clause]]
+==== VALUES Clause
+
+[subs="normal"]
+----
+values-clause ::= VALUES "(" _key_ "," _value_ ")" [ "," [ VALUES ] "(" _key_ "," _value_ ")" ]{asterisk}
+----
+
+image::n1ql-language-reference/values-clause.png["'VALUES' '(' key ',' value ')' ( ',' 'VALUES'? '(' key ',' value ')' )*"]
+
+key::
+A string, or an expression resolving to a string, representing the ID of the document to be inserted.
+The KEY cannot be MISSING or NULL, and must be unique within the Couchbase bucket.
 It can be a string or an expression that produces a string.
 
-You can specify a key using an expression, as shown here:
+value::
+A well-formed JSON object, or an expression resolving to a well-formed JSON object, representing the body of the document to be inserted.
+(See http://json.org/example.html[^] for examples of well-formed JSON.)
+You can insert NULL, empty, or MISSING values.
+
+****
+[.status]#Couchbase Server 6.5.1#
+
+In Couchbase Server 6.5.1, the VALUES clause supports an optional third argument, `__options__`.
+
+[subs="normal"]
+----
+values-clause ::= VALUES "(" _key_ "," _value_ [ "," _options_ ] ")" [ "," [ VALUES ] "(" _key_ "," _value_ [ "," _options_ ] ")" ]{asterisk}
+----
+
+image::n1ql-language-reference/values-clause-651.png["'VALUES' '(' key ',' value ( ',' options )? ')' ( ',' 'VALUES'? '(' key ',' value ( ',' options )? ')' )*"]
+
+options::
+[Optional] An object representing the metadata to be set for the inserted document.
+This object may only contain the `expiration` attribute.
```

Comment:
It can contain anything. Only "expiration" field from object is considered at present all others will be ignored.